scram-project-build.file

```diff
@@ -312,6 +312,9 @@ find external/%cmsplatf -type l | xargs ls -l
 %{?PatchReleaseSymlinkRelocate:%PatchReleaseSymlinkRelocate}
 echo "%{cmsroot}" > %{i}/config/scram_basedir
 
+# install checksum tool for local use (via scram)
+cp data/cmsTritonChecksumTool %i/bin/
```

Comment:
@smuzaffar is this the right way to install the tool into CMSSW? Running a full build to test this is a bit beyond my current resources...

Reply:
@kpedro88 , `scram-project-build` is used to build scram based project e.g. coral/cmssw etc. If this tool need to go in to cmssw/bin directory then why not add it under some package in cms-sw/cmssw repo?

Reply:
ah ok @kpedro88 , after reading https://github.com/cms-sw/cmssw/issues/43695 I remember where this tool is used. It should be copied here in scram-project-build.file. 

This tool is needed by cms-data externals. Can you please remind me how to run this for a cms-data repo? All cms-data external use https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_14_0_X/master/cmsswdata-github.file . So for selected cms-data externals (where we want to run it), we need to create a file e.g. `cmsdist/data/data-HeterogeneousCore-SonicTriton.file` with contents like
```
%define PostBuild %{cmsdist_directory}/data/cmsTritonChecksumTool <args>
```

so that ` %{cmsdist_directory}/data/cmsTritonChecksumTool ` can be run [here](https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_14_0_X/master/cmsswdata-github.file#L30)

Reply:
Yes, it needs to be here so it can run on cms-data packages during build time, and will then be copied into CMSSW. If there's a better place to execute that copy operation, I'm happy to move it.

My untested attempt to execute the tool during build time is here: (not in this PR, because I want to have everything in place before we start enforcing it)
https://github.com/kpedro88/cmsdist/commit/44f13f27f211d3a0e737e321a4696e8a26e75f32

This will be a no-op for most cms-data packages that don't have any config.pbtxt files, but using `find` avoids having to maintain a manual list (which is just one more thing to synchronize, i.e. one more potential source of failures when people forget to synchronize it).

Reply:
thanks @kpedro88 , about https://github.com/kpedro88/cmsdist/commit/44f13f27f211d3a0e737e321a4696e8a26e75f32, we should not modify the content of cmsswdata-github.file file otherwise it will rebuild all the data externals. We can add
```
%define PostBuild for CONFIG in $(find . -name config.pbtxt); do %{cmsdist_directory}/data/cmsTritonChecksumTool -c $CONFIG; done
```

in the `cmsdist/data/data-<Package-Name>.file` for only those data externals which contain `config.pbtxt`

Reply:
My concern, though, is that this is an additional manual step that has to be taken any time a Triton model is added to a (new or existing) cms-data package. This build-time check should be automatic. Rebuilding all the data externals would only happen once, right? Is that such a big problem?

Should I move the line from the "Prep" step to the "Build" step either way?

Reply:
yes rebuilding all data means re-distributing 20GB of data again. Although this is not a concern for CVMFS distribution but for online where cmssw is install on local disk this can cause issues.

Adding new data external is automated, bot know how to add a new data external in cmsdist. We can teach bot to create this cmsdist/data/data-<Package-Name>.file for new data external if it contains config.pbtxt

Reply:
@kpedro88 , see https://github.com/cms-sw/cms-bot/pull/2143

Reply:
@smuzaffar thanks! This looks like a good approach that satisfies both of our concerns.

A few questions:
1. the call to `cmsTritonChecksumTool -c $CONFIG` will go in the file `data/cmsTritonPostBuild.file`, once we're ready to enable it?
2. will this automatically propagate to the existing cms-data repos that contain config.pbtxt files, or do we have to add the `data-X-Y.file` files for those manually in cmsdist?

Reply:
> 1. the call to cmsTritonChecksumTool -c $CONFIG will go in the file data/cmsTritonPostBuild.file, once we're ready to enable it?

Yes

> will this automatically propagate to the existing cms-data repos that contain config.pbtxt files, or do we have to add the data-X-Y.file files for those manually in cmsdist?

No , not by default. `data-X-Y.file` will be created only when those data repos opens a new PR.